packages/api/src/command/medical/patient/converage-assessment-create.ts

```diff
@@ -22,38 +23,89 @@ export async function createCoverageAssessments({
   facilityId: string;
   patientCreates: PatientCreateCmd[];
 }): Promise<void> {
-  const chunkSize = 50;
+  const { log } = out(`createCoverageAssessments - cxId ${cxId}`);
   const patients: Patient[] = [];
 
-  const pdChunks = chunk(patientCreates, chunkSize);
-  for (const chunk of pdChunks) {
-    await sleep(delayTime.asMilliseconds());
-    const createPatients: Promise<Patient>[] = [];
-    for (const patient of chunk) {
-      createPatients.push(createOrUpdatePatient(patient));
+  async function createOrUpdatePatientWrapper({
+    patientCreateCmd,
+    patients,
+    log,
+  }: {
+    patientCreateCmd: PatientCreateCmd;
+    patients: Patient[];
+    log: typeof console.log;
+  }): Promise<void> {
+    try {
+      const patient = await createOrUpdatePatient(patientCreateCmd);
+      patients.push(patient);
+    } catch (error) {
+      const msg = `Failed to create or update patient. Cause: ${errorToString(error)}`;
+      log(msg);
+      capture.error(msg, {
+        extra: {
+          cxId,
+          facilityId,
+          context: "coverage-assessment.create",
+        },
+      });
     }
-    const patientResults = await Promise.allSettled(createPatients);
-    patientResults.map(result => {
-      if (result.status == "fulfilled") patients.push(result.value);
-    });
   }
 
-  const dQChunks = chunk(patients, chunkSize);
-  for (const chunk of dQChunks) {
-    await sleep(delayTime.asMilliseconds());
-    const docQueries: Promise<DocumentQueryProgress>[] = [];
-    for (const patient of chunk) {
-      docQueries.push(
-        queryDocumentsAcrossHIEs({
+  await executeAsynchronously(
+    patientCreates.map(patientCreateCmd => {
+      return { patientCreateCmd, patients, log };
+    }),
+    createOrUpdatePatientWrapper,
+    {
+      numberOfParallelExecutions: 50,
+      minJitterMillis: delayTime.asMilliseconds(),
+    }
+  );
+
+  async function queryDocumentsAcrossHIEsWrapper({
+    cxId,
+    patient,
+    facilityId,
+    log,
+  }: {
+    cxId: string;
+    patient: Patient;
+    facilityId: string;
+    log: typeof console.log;
+  }): Promise<void> {
+    try {
+      await queryDocumentsAcrossHIEs({
+        cxId,
+        patientId: patient.id,
+        facilityId,
+        triggerConsolidated: true,
+      });
+    } catch (error) {
+      const msg = `Failed query docuemnts for patient ${patient.id}. Cause: ${errorToString(
+        error
+      )}`;
+      log(msg);
+      capture.error(msg, {
+        extra: {
           cxId,
-          patientId: patient.id,
           facilityId,
-          triggerConsolidated: true,
-        })
-      );
+          patientId: patient.id,
+          context: "coverage-assessment.create",
+        },
+      });
```

Comment:
ditto: capture outside of the loop/wrapper

Reply:
Done.